AI 식물 추천 페이지 UI 구현

gayeon4679 opened the pull request:
## 📝 설명
AI 식물 추천 페이지 UI를 구현했습니다.
<!-- PR에 대한 설명입니다. -->

## ✅ PR 유형

- [x]  UI  디자인 구현

## 💻 작업 내용
## 🎨UI
![image](https://github.com/user-attachments/assets/5ff5ed32-9ce0-4ed1-a227-17179d314f9a)
![image](https://github.com/user-attachments/assets/dab6a651-56bf-48d4-bd6a-35e81723016f)
![image](https://github.com/user-attachments/assets/f76046a9-cff4-4e19-b3e1-1c0d4f5f78fc)
![image](https://github.com/user-attachments/assets/2a19c1ad-6ba6-48ae-b7c8-cc232a80ea2a)


<!-- PR 본문을 입력하세요. -->

## 💬 PR 포인트
- AIRecommend.tsx에서 input 값이 onChange 이벤트로 변경될 때마다 useState를 이용해 값을 저장하는 방식으로 작성하였습니다. 일반적으로 이런 방식을 쓰는 것 같아서 이렇게 작성하였는데 혹시 이런 경우에는 useRef를 쓰는게 더 적합한지 궁금합니다.

- 파일들이 폴더 구조대로 잘 되어 있는지 확인해주세요!


<!-- PR 리뷰 시 공유 사항 또는 유심히 보면 좋을 부분을 설명합니다. -->

gayeon4679 committed:
[🎨design]: AI 식물 추천 페이지 UI 구현

gayeon4679 committed:
Merge branch 'dev' of https://github.com/High-Coding-org/High-Coding-FE into Design/#54-AiPlantRecomandPage

meowTarae commented on src/pages/MyPlantPage/PlantRecommend/PlantRecommend.tsx:
```diff
@@ -1,3 +1,64 @@
+  };
+
+  return (
+    <main className="w-full px-5 md:w-[1140px]">
```
main 태그 안에 Breadcrumb과 전체 내용이 있는 것 보다는  Breadcrumb이 main 태그 밖에 있는건 어떨까요?
fragment로 Breadcrumb과 main를 감싸는거죠.
왜냐하면 Breadcrumb이 해당 추천 페이지에 주 요소가 아니라고 생각하기 때문이에요.

gayeon4679 replied:
저번에 얘기해주셨는데 이게 푸쉬가 안되어있더라고요😭 수정해서 올리겠습니다!

meowTarae commented on src/pages/MyPlantPage/PlantRecommend/PlantRecommend.tsx:
```diff
@@ -1,3 +1,64 @@
+            className="flex flex-col gap-8"
+            onSubmit={handleSubmit}>
+            <div className="flex flex-col gap-4">
+              <div className="flex flex-col gap-2">
+                <Label htmlFor="temperature">온도(°C)</Label>
+                <Input
+                  type="number"
+                  id="temperature"
+                />
+              </div>
```
중복되는 코드가 보이네요! 해당 코드는 리팩토링 해주면 좋을 것 같습니다.

저라면 이렇게 리팩토링 해볼 것 같아요.
```
// 컴포넌트 파일로 분리
export const Container = ({ label, id, type }: { label: string; id: string; type: string }) => {
  return (
    <div className="flex flex-col gap-2">
      <Label htmlFor={id}>{label}</Label>
      <Input
        type={type}
        id={id}
      />
    </div>
  );
};

// PlantRecommend.tsx
const CURRENT_ENVIRONMENT = [
  {
    label: '온도(°C)',
    id: 'temperature',
    type: 'number',
  },
  {
    label: '습도(%)',
    id: 'humidity',
    type: 'number',
  },
  ...
];

CURRENT_ENVIRONMENT.map((item, idx) => (
  <Container
    key={idx}
    label={item.label}
    id={item.id}
    type={item.type}
  />
));

```

하드코딩된 텍스트들을 상수화 할 수 있기도 하네요~

gayeon4679 replied:
중복코드를 없애는 방식으로 리팩토링하였습니다🫡

meowTarae commented on src/pages/MyPlantPage/PlantRecommend/PlantRecommend.tsx:
```diff
@@ -1,3 +1,64 @@
+    <main className="w-full px-5 md:w-[1140px]">
+      {/*<BreadcrumbAndTitle />*/}
+
+      <section className="bg-[#F5F5F5] flex flex-col justify-between gap-8 p-12 rounded h-auto md:flex-row md:gap-20">
```
식물 추천 영역에 대한 구분을 위해 색상을 잘 부여해주셨는데요, input창까지 색상을 부여하면 유저 입장에서 인식이 조금 어려울 수 있을 것 같아요.

gayeon4679 replied:
그렇네요. input 창의 색상은 변경하도록 하겠습니다!

meowTarae commented on src/pages/MyPlantPage/PlantRecommend/PlantRecommend.tsx:
```diff
@@ -1,3 +1,64 @@
+              <div className="flex flex-col gap-2">
+                <Label htmlFor="temperature">온도(°C)</Label>
+                <Input
+                  type="number"
```
입력받는 값이 숫자 값이고, 값들이 편차가 그리 심하지않아 Input type을  number로 설정해두신 부분이 좋은 것 같아요 👍 

그런데 온도 같은 경우 0.1 단위로 늘렸다 줄였다 하는 부분이 많을 것 같은데, 그런 부분에 있어 버튼 클릭 시 0.1 단위로 증감시키는 기능은 혹시 어떻게 생각하실까요 ?

gayeon4679 replied:
좋은 생각인 것 같아요😯

meowTarae commented on src/pages/MyPlantPage/PlantRecommend/PlantRecommend.tsx:
```diff
@@ -1,3 +1,64 @@
+//! Breadcrumb 추가해야 함
 export default function PlantRecommend() {
-  return <div>식물 추천해주는 페이지</div>;
+  const handleSubmit = (e: React.FormEvent) => {
```
요런 함수명은 추후 기능이 많아질 때 혼돈을 불러올 수 있기에, 이름을 명확히 해주시면 더 좋을 것 같아요.

gayeon4679 replied:
넵 수정하겠습니다!

meowTarae reviewed (approved):
피그마에 있는 내용처럼 잘 만들어주셨습니다. 고생 많으셨어요!
그런데 pr 올리기 전, gpt 등을 사용해서 가볍게 코드 퀄리티 체크 후 올려주시면 더 좋을 것 같아요.

gayeon4679 committed:
[✨feat]: lottie 식물 json 파일 추가

gayeon4679 committed:
[♻️refactor]: lottie 애니메이션 추가, 전체 내용을 감싼 태그 main에서 fragment로 변경

gayeon4679 committed:
Merge branch 'dev' of https://github.com/High-Coding-org/High-Coding-FE into Design/#54-AiPlantRecomandPage

gayeon4679 committed:
[♻️refactor]: 각 컴포넌트를 개별 파일로 분리

gayeon4679 committed:
[♻️refactor]: ENVIRONMENT_FIELDS 상수 별도 파일로 분리

gayeon4679 committed:
[🚚rename]: plantRecommend 이름 AIRecommend로 변경

gayeon4679 committed:
[💄style]: tailwind CSS 순서 정리

gayeon4679 committed:
[🔨chore]: react-svg-radar-chart 설치

gayeon4679 committed:
[🎨design]: ai 식물추천페이지 수정된 ui에 맞춰 변경

gayeon4679 committed:
Merge branch 'dev' of https://github.com/High-Coding-org/High-Coding-FE into Design/#54-AiPlantRecomandPage

gayeon4679 committed:
[🎨design]: AiPlantPage 디자인 수정

SliderContainer 윗 부분에 제목 추가, ChevronLeft와 ChevronRight 좌우에 영역 설정하고 크기 키움, form 옆에 구분선 추가
